UI, MapLocation, UI for Store Info

ilijaS opened the pull request:
-implemented UI for dispalying map location of store
-created UI for displaying store info
-refined UI for displaying stores in grid

ilijaS committed:
UI, MapLocation, UI for Store Info

-implemented UI for dispalying map location of store
-created UI for displaying store info
-refined UI for displaying stores in grid

ilijaS committed:
Merge branch 'StoreLocatorRedone' into UIStoreInformation

marko-lf commented:
In InitVC, when we switch to real data, we are still waiting for random time to be switched to store list screen. We want to get there asap.

Better solution is to completely remove Init screen since there is no real need for it.
Location can be retrieved in store list screen and in meanwhile stores can be displayed or not. If they are displayed, instead of distance there can be an activity indicator spinning until location is retrieved. Other option can show activity indicator in front of empty screen and when location is ready spinner is removed and all data is displayed.

Take a look at CLLocationManagerDelegate for available callbacks when you request a location from LocationManager.

=========================================
StoreVC line 88
    for store in (model?.stores)!
...
It's cleaner to have:
func storeForID(_ storeID:Int) -> [Store] 

Especially when we find store to be the first in model?.stores and all remaining iterations are unnecessary.

=========================================

public class storeModel:
Class names should start with capital letter: "StoreModel" ("Store" is even better name because classes with name like "Store" means it's model in MVC - if it's some view displaying store than its "StoreView" or VC controlling those views -> "StoreVC")

=========================================

Network communication module shouldn't be part of the StoreModel. Use Alamofire lib.
Random generator shouldn't be part of the StoreModel.
Location manager shouldn't be part of the StoreModel.
Call Store  shouldn't be part of the StoreModel. (unless its callStore(storeID), then inside callStore with given storeID should be call(number) and call should be function defined outside StoreModel.

marko-lf reviewed (changes requested):
feedback in comment above ^
:)

ilijaS committed:
Alamofire integrated

-alamofire integrated
-store module split into store module, network module and location module

ilijaS committed:
more pull request feedback tweaks

-implemented UI for store detail display
-removed miscellaneous functionalities (call the store, RNG) from the store model
-changed the name of model classes

ilijaS committed:
Cell - Store Name label overlapping issue fixed (!?)

Setting the store name label 'opaque' attribute to 'non-opaque' seems to completely fix the issue of the cell store name overlapping.
However, this doesn't feel like the propper way to deal with this problem, more like a 'quick and dirty' solution.

ilijaS committed:
NVActivityIndicator integrated

pull to refresh now looks way more fancy

ilijaS committed:
System Alert Views for errors

Most use-cases when error could occur now open a system alert view window instead of just printing the string describing an error in the debug console

ilijaS committed:
Commit for the review

Cleaned and restructured the code from the previous commit

ilijaS committed:
No internet connection use case  covered

Changes in this commit:

-Made the app usable in the scenario when there is no internet connection. An alert popup appears that informs the user that there is no connection available.

-Made changes to the way the store is fetched in the view as it was suggested in the last pull request

-Added the cell border options that were set in the project requirements
